test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/CompositeModelBinderTest.cs

```diff
@@ -321,56 +296,28 @@ public class CompositeModelBinderTest
                 FieldName = "someName",
             };
 
-            var modelBinder = new Mock<IModelBinder>();
-            modelBinder
-                .Setup(mb => mb.BindModelAsync(It.IsAny<ModelBindingContext>()))
-                .Returns(ModelBindingResult.SuccessAsync("someName", model: null));
+            var mockIntBinder = new StubModelBinder(context =>
+            {
+                context.Result = ModelBindingResult.Success("someName", model: null);
+            });
 
-            var composite = CreateCompositeBinder(modelBinder.Object);
+            var composite = CreateCompositeBinder(mockIntBinder);
 
             // Act
-            var result = await composite.BindModelAsync(bindingContext);
+            var result = await composite.BindModelResultAsync(bindingContext);
 
             // Assert
-            Assert.NotEqual(ModelBindingResult.NoResult, result);
+            Assert.NotEqual(default(ModelBindingResult), result);
             Assert.True(result.IsModelSet);
             Assert.Equal("someName", result.Key);
             Assert.Null(result.Model);
         }
 
         [Fact]
-        public async Task BindModel_UnsuccessfulBind_BinderFails_ReturnsNoResult()
```

Comment:
Good to see `BindModel_UnsuccessfulBind_BinderFails_ReturnsNoResult()` go away. Was an artifact from time when top-level simple types were implicitly `[BindRequired]` (or maybe `[Required]`).